archive/finder.go

```diff
@@ -0,0 +1,68 @@
+package archive
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/brimsec/zq/pkg/sst"
+)
+
+func Find(dir string, pattern []byte) ([]string, error) {
+	nerr := 0
+	//XXX this should be parallelized with some locking presuming a little
+	// parallelism won't mess up the file system assumptions
+	var hits []string
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("%q: %v", path, err)
+		}
+		name := info.Name()
+		if info.IsDir() {
+			if filepath.Ext(name) == zarExt {
+				//XXX need to merge into or replace existing index
+				return filepath.SkipDir
+			}
+			// descend...
+			return nil
+		}
+		if filepath.Ext(name) == ".bzng" {
+			hit, err := SearchFile(path, pattern)
+			if err != nil {
+				fmt.Printf("%s\n", err)
+				nerr++
```

Comment:
Even in this prototype, why skip over any errors?

Reply:
It's not skipping errors.  It was bailing after too many errors.  I'll take it out.